src/services/SearchResults.cpp

```diff
@@ -234,17 +234,18 @@ void SearchResults::ProcessBlocksNibbles(const SearchResults& srSource, unsigned
     for (auto& block : srSource.m_vBlocks)
     {
         if (block.GetSize() > vMemory.capacity())
-            vMemory.reserve(block.GetSize());
+            vMemory.resize(block.GetSize());
 
-        unsigned char* pMemory = vMemory.data();
-        const unsigned char* pPrev = block.GetBytes();
+        std::array<unsigned char, sizeof(MemBlock)> pPrev{};
+        for (auto i = 0U; i < sizeof(MemBlock); i++)
+            pPrev.at(i) = block.GetBytes()[i];
```

Comment:
This was the only reasonable way I could do this, if it's still not good enough, we might have to get the Spectre Mitigated compilers and libraries. If there's a better way to do this without resorting to that, please come forward.

Reply:
There has to be a better way. There's absolutely no reason to copy the data just to read it. Maybe instead of storing a pointer to the backing array, add a `GetByte(index)` method.